pkg/openservicebroker/route.go

```diff
@@ -0,0 +1,150 @@
+package openservicebroker
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	restful "github.com/emicklei/go-restful"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+// minimum supported client version
+const minAPIVersionMajor, minAPIVersionMinor = 2, 7
+
+func Route(container *restful.Container, path string, b Broker) {
+	shim := func(f func(Broker, *restful.Request) *Response) func(*restful.Request, *restful.Response) {
+		return func(req *restful.Request, resp *restful.Response) {
+			response := f(b, req)
+			if response.Err != nil {
+				resp.WriteHeaderAndJson(response.Code, &ErrorResponse{Description: response.Err.Error()}, restful.MIME_JSON)
+			} else {
+				resp.WriteHeaderAndJson(response.Code, response.Body, restful.MIME_JSON)
+			}
+		}
+	}
+
+	ws := restful.WebService{}
+	ws.Path(path + "/v2")
+	ws.Consumes(restful.MIME_JSON)
+	ws.Produces(restful.MIME_JSON)
+	ws.Filter(apiVersion)
+
+	ws.Route(ws.GET("/catalog").To(shim(catalog)))
+	ws.Route(ws.PUT("/service_instances/{instance_id}").To(shim(provision)))
+	ws.Route(ws.DELETE("/service_instances/{instance_id}").To(shim(deprovision)))
+	ws.Route(ws.GET("/service_instances/{instance_id}/last_operation").To(shim(lastOperation)))
+	ws.Route(ws.PUT("/service_instances/{instance_id}/service_bindings/{binding_id}").To(shim(bind)))
+	ws.Route(ws.DELETE("/service_instances/{instance_id}/service_bindings/{binding_id}").To(shim(unbind)))
```

Comment:
yes i think we're going to want a specific set of SC credentials used to call this, and not give the SC system:admin credentials.